source/common/uma_pools/disjoint_pool.cpp

```diff
@@ -67,6 +67,10 @@ static void *AlignPtrUp(void *Ptr, const size_t Alignment) {
     return static_cast<char *>(AlignedPtr) + Alignment;
 }
 
+struct MemoryProviderError {
+    uma_result_t code;
```

Comment:
LGTM
nit: perhaps a better name for this variable would be retCode? I think it may be more descriptive.